File: practicas/practica_2/Deepspace_Ruby_P2/lib/TestP2.rb
#encoding:utf-8

# File: TestP2.rb
# Author: Juan Manuel Rodríguez Gómez

require_relative 'Hangar'
require_relative 'ShieldBooster'
require_relative 'Weapon'
require_relative 'WeaponType'
require_relative 'Damage'
require_relative 'Loot'
require_relative 'EnemyStarShip'

module Deepspace
  class TestP2
    
    def self.main
      puts "Iniciando TestP2"

      hangar = Deepspace::Hangar.new(2)
      hangarCopy = Deepspace::Hangar.newCopy(hangar)
      shield = Deepspace::ShieldBooster.new("Shield", 2.4, 3)
      laser = Deepspace::Weapon.new("Laser", Deepspace::WeaponType::LASER, 2)
      plasma = Deepspace::Weapon.new("Plasma", Deepspace::WeaponType::PLASMA, 4)
      missile = Deepspace::Weapon.new("Missile", Deepspace::WeaponType::MISSILE, 3)
      
      weaponTypes = [Deepspace::WeaponType::LASER, Deepspace::WeaponType::PLASMA]
      weaponTypes.push(Deepspace::WeaponType::PLASMA)
      weaponTypes.push(Deepspace::WeaponType::PLASMA)
      weapons = [laser, plasma]
      
      weapAdjust = [laser, missile]
      shAdjust = [shield]
      
      damage = Deepspace::Damage.newNumericWeapons(5,3)
      damageSpecific = Deepspace::Damage.newSpecificWeapons(weaponTypes, 4)
      damageCopy = Deepspace::Damage.newCopy(damage)
      damageSpecificCopy = Deepspace::Damage.newCopy(damageSpecific)
      
      loot = Deepspace::Loot.new(4, 4, 5, 3, 6)
      enemy = Deepspace::EnemyStarShip.new("Enemigo", 2, 3, loot, damage)
      enemyCopy = Deepspace::EnemyStarShip.newCopy(enemy)
      
      # HANGAR
      puts "\n----- HANGAR -----\n"
      puts "Hangar ->\n%s" % hangar.to_s
      puts "\nHangarCopy ->\n%s" % hangarCopy.to_s
      
      puts "\nAñadiendo un escudo al original y un laser al hangar original y a la copia...\n"
      hangar.addShieldBooster(shield)
      hangar.addWeapon(laser)
      hangarCopy.addWeapon(laser)
      puts "Hangar ->\n%s" % hangar.to_s
      puts "\nHangarCopy ->\n%s" % hangarCopy.to_s
      
      puts "\nHangar ->\n%s" % hangar.to_s
      puts "\nHangarCopy ->\n%s" % hangarCopy.to_s

      puts "\nEliminando escudo del original...\n"
      hangar.removeShieldBooster(0)
      puts "\nHangar ->\n%s" % hangar.to_s
      puts "\nHangarCopy ->\n%s" % hangarCopy.to_s
      
      puts "\nEliminando primera arma de la copia...\n"
      hangarCopy.removeWeapon(0)
      puts "\nHangar ->\n%s" % hangar.to_s
      puts "\nHangarCopy ->\n%s" % hangarCopy.to_s
    
      # DAMAGE
      puts "\n----- DAMAGE -----\n"
      puts "Damage ->\n%s" % damage.to_s
      puts "\nDamageSpecific ->\n%s" % damageSpecific.to_s
      puts "\nDamageCopy ->\n%s" % damageCopy.to_s
      puts "\nDamageSpecificCopy ->\n%s" % damageSpecificCopy.to_s

      puts "\nAjustando DamageSpecific..."
      adjust = damageSpecific.adjust(weapAdjust, shAdjust)
      puts "\nDamageSpecific ->\n%s" % damageSpecific.to_s
      puts "\nDamageSpecificAdjust ->\n%s" % adjust.to_s
      
      puts "\nDescartando PLASMA y un escudo..."
      damage.discardWeapon(plasma)
      damageSpecific.discardWeapon(plasma)
      damage.discardShieldBooster
      damageSpecific.discardShieldBooster
      puts "Damage ->\n%s" % damage.to_s
      puts "\nDamageSpecific ->\n%s" % damageSpecific.to_s
      puts "\nDamageCopy ->\n%s" % damageCopy.to_s
      puts "\nDamageSpecificCopy ->\n%s" % damageSpecificCopy.to_s
      
    end
  end # Class

  TestP2.main

end # Module

File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/TestP2.java
/**
 * @file TestP2.java
 * @author Juan Manuel Rodríguez Gómez
 */
package deepspace;

import java.util.ArrayList;

/**
 * @brief Test Deepspace Práctica 2
 */
public class TestP2 {
    
    public static void main(String[] args) {
        final String LINE = "\n=========================\n";
        
        Hangar hangar = new Hangar(2);
        Hangar hangarCopy =new Hangar(hangar);
        ShieldBooster shield = new ShieldBooster("Shield", 2.4f, 3);
        Weapon laser = new Weapon("Laser", WeaponType.LASER, 2);
        Weapon plasma = new Weapon("Plasma", WeaponType.PLASMA, 4);
        Weapon missile = new Weapon("Missile", WeaponType.MISSILE, 3); 
        
        ArrayList<WeaponType> weaponTypes = new ArrayList<>();
        weaponTypes.add(WeaponType.LASER);
        weaponTypes.add(WeaponType.PLASMA);
        weaponTypes.add(WeaponType.PLASMA);
        weaponTypes.add(WeaponType.PLASMA);
        
        ArrayList<Weapon> weapons = new ArrayList<>();
        weapons.add(laser);
        weapons.add(plasma);
        
        ArrayList<Weapon> weapAdjust = new ArrayList<>();
        weapAdjust.add(laser);
        weapAdjust.add(missile);
        ArrayList<ShieldBooster> shAdjust = new ArrayList<>();
        shAdjust.add(shield);
        
        
        Damage damage = new Damage(5, 3);
        Damage damageSpecific = new Damage(weaponTypes, 4);
        Damage damageCopy = new Damage(damage);
        Damage damageSpecificCopy = new Damage(damageSpecific);
        
        Loot loot = new Loot(4,4,5,3,6);
        EnemyStarShip enemy = new EnemyStarShip("Enemigo", 2, 3, loot, damage);
        EnemyStarShip enemyCopy = new EnemyStarShip(enemy);
        
        // HANGAR
        System.out.println("\n----- HANGAR -----\n");
        System.out.printf("Hangar ->\n%s\n", hangar.toString());
        System.out.printf("\nHangarCopy ->\n%s\n", hangarCopy.toString());
        
        System.out.println("\nAñadiendo un escudo al original y un laser al hangar original y a la copia...\n");
        hangar.addShieldBooster(shield);
        hangar.addWeapon(laser);
        hangarCopy.addWeapon(laser);      
        System.out.printf("Hangar ->\n%s\n", hangar.toString());
        System.out.printf("\nHangarCopy ->\n%s\n", hangarCopy.toString());
        
        System.out.println("\nEliminando escudo del original...\n");
        hangar.removeShieldBooster(0);      
        System.out.printf("Hangar ->\n%s\n", hangar.toString());
        System.out.printf("\nHangarCopy ->\n%s\n", hangarCopy.toString());

        System.out.println("\nEliminando arma de la copia...\n");
        hangarCopy.removeWeapon(0);     
        System.out.printf("Hangar ->\n%s\n", hangar.toString());
        System.out.printf("\nHangarCopy ->\n%s\n", hangarCopy.toString());
        
        // DAMAGE
        System.out.println("\n----- DAMAGE -----\n");
        System.out.printf("Damage ->\n%s\n", damage.toString());
        System.out.printf("\nDamageCopy ->\n%s\n", damageCopy.toString());
        System.out.printf("\nDamageSpecific ->\n%s\n", damageSpecific.toString());
        System.out.printf("\nDamageSpecificCopy ->\n%s\n", damageSpecificCopy.toString());

        System.out.println("\nAjustando DamageSpecific...");
        Damage adjust = damageSpecific.adjust(weapAdjust, shAdjust);
        System.out.printf("\nDamageSpecific ->\n%s\n", damageSpecific.toString());
        System.out.printf("\nDamageSpecificAdjust ->\n%s\n", adjust.toString());
    
        System.out.println("\nDescartando PLASMA y escudo...");
        damage.discardWeapon(plasma);
        damageSpecific.discardWeapon(plasma);
        damage.discardShieldBooster();
        damageSpecific.discardShieldBooster();
        System.out.printf("Damage ->\n%s\n", damage.toString());
        System.out.printf("\nDamageCopy ->\n%s\n", damageCopy.toString());
        System.out.printf("\nDamageSpecific ->\n%s\n", damageSpecific.toString());
        System.out.printf("\nDamageSpecificCopy ->\n%s\n", damageSpecificCopy.toString());
    
        // ENEMYSTARSHIP
        System.out.println("\n----- ENEMY STARSHIP -----\n");
        System.out.printf("Enemy ->\n%s\n", enemy.toString());
        System.out.printf("\nEnemyCopy ->\n%s\n", enemyCopy.toString());
    }
}